core/trino-main/src/main/java/io/trino/execution/UseTask.java

```diff
@@ -62,7 +63,10 @@ public String getName()
             throw new TrinoException(NOT_FOUND, "Catalog does not exist: " + catalog);
         }
 
-        String schema = statement.getSchema().getValue().toLowerCase(ENGLISH);
+        Identifier schemaIdentifier = statement.getSchema();
+
+        String schema = metadata.getNameCanonicalizer(session, catalog)
```

Comment:
`metadata.canonicalize(session, catalog, schemaIdentifier.getValue(), schemaIdentifier.isDelimited());`?

Reply:
`NameCanonicalizer` abstraction allows to have two implementation one for engine and one based on connector.. Any issues in using this abstraction.. It allows us to toggle the implementation between the legacy and current canonicalization in the future.